doc/user-stories/expert-partitioner.md

```diff
@@ -0,0 +1,414 @@
+    * select partition type
+    * select partition size
+    * select partition role
+    * set partition attributes (fs, mount point, etc)
+
+### When we are in wizard step to select the partition type
+* and partition table is not MSDOS
+  * the step is skipped
+* and partition table is MSDOS
+  * and only it is possible to create primary partitions
+    * the step is skipped
+  * and only it is possible to create logical partitions
+    * the step is skipped
+  * and there is space for a primary partition
+    * and the max of primary partitions is not reached
+      * shows option 'Primary partition'
+  * and there is not an extended partition
+    * shows option 'Extended partition' 
+  * and there is an extended partition with free space
+    * shows option 'Logical partition'
+
+### When we are in wizard step to select the partition size
+* shows three options
+  * Maximun size
+  * Custom size
+    * with maximum size by default
+  * Custom region
+    * with start and end cylinders of the free region by default
+* *(pending) and partition is primary*
+  * custom size is selected by default
+* and partition is extended
+  * maximum size is selected by default
+* *(pending) and partition is logical*
+  * custom region is selected by default
+* and 'Next' is selected
+  * and partition is extended
+    * finishes wizard
+    * creates the partition
+
+### When we are in wizard step to select the role of the partition
+* shows the following options
+  * operating system (i.e. root)
+  * data and ISV applications (any other mount point)
+  * swap
+  * raw volume (unformatted)
+* option 'data and ISV applications' is selected by default
+
+### When we are in wizard step for partition attributes
+* shows two sets of options: 'Formatting options' and 'Mounting options'
+  * where 'Formatting options' contains
+    * Format partition
+    * Do not format partition
+    * Encrypt device
+  * where 'Mounting options' contains
+    * Mount partition
+    * Do not mount partition
+* *(pending) and partition role is 'Operating system'*
+  * sets the following 'Formatting options' by default
+    * Format partition: true
+      * File System: Btrfs
+      * Enable snapshots: true
+    * Do not format partition: false
+      * File System Id: 0x83 Linux
+    * Encrypt: false
+  * sets the following options for 'Mounting options'
+    * Mount partition: true
+      * Mount point: first free of /, /home, /var
+    * Do not mount partition: false
+* *(pending) and partition role is 'Data and ISV applications'*
+  * sets the following 'Formatting options' by default
+    * Format partition: true
+      * File System: XFS
+    * Do not format partition: false
+      * File System Id: 0x83 Linux
+    * Encrypt: false
+  * sets the following options for 'Mounting options'
+    * Mount partition: true
+      * Mount point: first free of /, /home, /var
+    * Do not mount partition: false
+* *(pending) and partition role is 'Swap'*
```

Comment:
In the new version, when you select role 'Swap', then in the next step the options "Format device" and "Mount Device" are not selected by default. Ok, I will set as pending only this two options.